app/main.py

```diff
@@ -19,16 +23,27 @@
     allow_headers=["*"],
 )
 
+limiter = Limiter(
+    key_func=get_remote_address,
+    default_limits=[settings.RATE_LIMIT],
+    enabled=settings.PYTHON_ENV == "production",
+)
+app.state.limiter = limiter
+app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
+app.add_middleware(SlowAPIMiddleware)
```

Comment:
Can you explain a bit abt how this works? Im lazy to go read `slowapi` document

Reply:
Create an app-wide instance of python `collections.Counter` to store client connected, with key got by `get_remote_address()` and value is number of request in the time window. If value exceed `settings.RATE_LIMIT`, return 429 error.